test: add coverage for spawnSync() killSignal

cjihrig opened the pull request:
##### Checklist
- [x] `make -j8 test` (UNIX), or `vcbuild test nosign` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
test

##### Description of change
This commit adds a test for the `killSignal` option to `spawnSync()`, and the other sync child process functions by extension.

This was previously untested according to https://node-core-coverage.addaleax.net/

bnoordhuis commented on test/parallel/test-child-process-spawnsync-kill-signal.js:
```diff
@@ -0,0 +1,50 @@
+if (process.argv[2] === 'child') {
+  setInterval(() => {}, 1000);
+} else {
+  const SIGUSR2 = process.binding('constants').os.signals.SIGUSR2;
```
You could DRY this as `const { SIGUSR2 } = process.binding('constants').os.signals`.  Minor thing though.

cjihrig commented:
CI with the suggestion incorporated: https://ci.nodejs.org/job/node-test-pull-request/4445/

cjihrig commented:
Related failures on Windows.

cjihrig commented:
It looks like `child.status` is 1 on Windows, and 0 elsewhere. Additionally, certain signals are not supported on Windows. Added detection of the exit code, and switched from `SIGUSR2` to `SIGKILL`.

CI: https://ci.nodejs.org/job/node-test-pull-request/4447/. The only failures are now unrelated.
